plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/NonSystemJobSuspender.java

```diff
@@ -45,41 +45,48 @@ private SuspendedJob(Job job, long scheduleDelay) {
     }
   }
 
+  private static final AtomicBoolean suspended = new AtomicBoolean(false);
+
   private static final List<SuspendedJob> suspendedJobs = new ArrayList<>();
 
   private static final NonSystemJobScheduleListener jobScheduleListener =
       new NonSystemJobScheduleListener();
 
   /** Once called, it is imperative to call {@link resume()} later. */
-  public static synchronized void suspendFutureJobs() {
-    Preconditions.checkState(!suspended, "Already suspended.");
-    suspended = true;
+  public static void suspendFutureJobs() {
```

Comment:
should this whole method be synchronized  on suspended (or some other shared object?) 

In particular, what if the state of suspended changes after getAndSet returns but before checkState executes?